### DISCUSSION TITLE
Sunsynk Dongle

### DISCUSSION THREAD

**Discussion by mikescooper** [kellerza/sunsynk#270] [General]
Hi,
I'm trying to configure your software to work with the Sunsynk Dongle. Before using the RS485 & RS232 connections (remove the Sunsynk dongle from RS232 port)
I have the mqtt software loaded and running, created the user for mqtt and set the password entered the details in the configuration part of the setup.
I've entered the serial number in for the inverter.
Entered the Port No found from the router. I assume the port has the e-linter as the device name.
Started the program and checked the log
**Result see below**
10:33:56 INFO    Using Single phase sensor definitions.
10:33:56 INFO    Added hidden sensors as other sensors depend on it: Rated power, Serial
10:33:56 INFO    Connecting to tcp://192.168.1:229
10:33:56 INFO    PyModbus 3.5.2 tcp: 192.168.1:229
10:33:59 WARNING Failed to connect 
10:33:59 INFO    ############################################################
10:33:59 INFO             Could not connect to tcp://192.168.1:229:
10:33:59 INFO    ############################################################
10:33:59 CRITICAL This Add-On will terminate in 30 seconds, use the Supervisor Watchdog to restart automatically.

Have I missed something awaiting your valued comments.
Regards
Mike

**Comment by kellerza** [comment]
`tcp://192.168.1:229:` is not a valid address?

Maybe it should be `tcp://192.168.1.229:8899` ?

**Comment by Nardus88** [comment]
Any follow up on this, did you get it right @mikescooper? I have the Sunsynk dongle and would prefer to use it if possible.

Otherwise I need to go the RS485 route, this will just be more convenient.

New to HA and integration, help will be useful!

Regards Nardus

**Comment by Waz-Cpt** [comment]
Hello Gents,

The E-Linter dongles are not supported by this integration with stock firmware.

You would need to flash a custom firmware to convert the dongle into a serial2net or Modbus Gateway.

I have previously posted a solution to use RS485 module and an ESP32 board to create a Modbus TCP Gateway. I have tested it and been running it without issues with this integration.

The Solarman dongle provides the required data stream from this integration to work when selecting the Solarman driver in the integration.

@kellerza, should I to a pull request to update information to include the Modbus Gateway project I used?

**Reply by jeybee74** [reply]
Hi,
I also have the E-linter dongle on my 12k inverter (3phase), I believe that my inverter allows read data from the RS485 port(s) and that Read+Write is only through the RS232 port (where the dongle is connected).

I hope it is possible to both use the Sunsynk app and HA for monitoring AND changing a few parameters like Charge A and SOC in the time of use settings with automations from HA.

Do you know that its possible to use the E-Linter dongle with Sunsynk system + HA with the custom firmware ?

Jon, Norway

**Reply by Waz-Cpt** [reply]
Hello Jon,

There is a Node-RED based cloud solution that uses the Sunsynk cloud to gather data and adjust settings. The problem with this approach is that Sunsynk restrict the polling rate to a min of 60s, however may users report they default everyone down to a min of 3 minutes and by default the system is actually at 5 minutes. The integration information can be found on [www.powerforum.co.za](url)

You could use that node-red process for changing settings and the KellerZa integration for reliable data gathering.

I have not seen any progress on the hacking and reverse engineering of the E-linter dongle to have a customer firmware to provides the data and firmware upgrade features of the stock unit while adding the access to the data on a local network.

You can maybe ask the guys at SolarAssistant for confirmation on whether the settings issues over RS485 port has been solved.

**Reply by Waz-Cpt** [reply]
Also to note, the RS232 port apparently just provides Modbus over serial.

If you do not need the Sunsynk cloud and can use home assistant for everything, then you can maybe purchase a second E-linter dongle and flash it with the custom ESPhome firmware and just connect the original dongle when you need to update firmware. You could also make your own dongle or use a USB to RS232 serial cable and connect it directly to your Home Assistant machine for use with the KellerZA integration. I believe you may also be able to use an off the shelf serial to Ethernet IP device.

**Reply by jeybee74** [reply]
Thanks for your reply!

I currently use this integration; [solarsynkv2](https://github.com/martinville/solarsynkv2) , but it is just fetching and sending data via the cloud, so if internet or the sunsynk cloud service is down there is holes in the logging and my automations stops working...

So direct comms would be much better! :) 

I have asked Deye if they can confirm if I can set parameters over RS485 and maybe they can fix this with a new firmware.

I guess my options are:
  1. get a second E-linter and update the firmware so I can use it directly, but then I loose all Sunsynk cloud functions right?
  2. get a solarman dongle, I guess this can work both locally and to Sunsynk cloud at once?
  3. If I can set parameters locally over RS485, I'll keep my dongle and just add communication and use this integration alongside the sunsynk cloud.
  4. Best option would be if someone could fix so the E-linter dongle works both locally and with the cloud at the same time, but that could not be possible..?

Thanks. 
Jon

**Reply by Waz-Cpt** [reply]
Martin's add-on has some flaws but it is simple and effective.

Gary's Node-RED flow solution allows for easier changes to settings but is more complicated for initial setup and config.

SolarAssistant is a really streamlined solution which can make use of a RPi Zero W2 and offers a cloud interface as well as local MQTT data. You can either use the USB to Serial adaptor to the logger port or RS485 dongle BMS port.

Flashing custom firmware on the dongle can provide direct Modbus integration in ESPhome or you can make use of a ESP32 Modbus Gateway project and KellerZa integration.

**Comment by mikescooper** [comment]
Using a different way of getting the information from the inverter.
